xds/src/main/java/io/grpc/xds/GoogleCloudToProdNameResolver.java

```diff
@@ -175,7 +175,7 @@ public void run() {
 
   private ImmutableMap<String, ?> generateBootstrap(String zone, boolean supportIpv6) {
     ImmutableMap.Builder<String, Object> nodeBuilder = ImmutableMap.builder();
-    nodeBuilder.put("id", "C2P-" + rand.nextInt());
+    nodeBuilder.put("id", "C2P-" + (rand.nextInt() & 0x7FFFFFFF));
```

Comment:
nit: Is `0x7FFFFFFF` `Integer.MAX_VALUE`? I have to carefully count how many `F`'s are there.